modules/placement-driver/src/main/java/org/apache/ignite/internal/placementdriver/leases/LeaseTracker.java

```diff
@@ -230,6 +239,49 @@ public void onError(Throwable e) {
         }
     }
 
+    /**
+     * Wait for a new primary replica in case when the current one is inoperable.
+     *
+     * @param groupId Group id.
+     * @param timestamp CLOCK_SKEW aware timestamp reference value.
+     * @param inoperableLeaseStartTime Start time of inoperable lease.
+     * @param timeout How long to wait before completing exceptionally with a TimeoutException, in milliseconds.
+     * @return New leaseholder future.
+     */
+    private CompletableFuture<ReplicaMeta> awaitNewPrimaryReplica(
```

Comment:
As mentioned above I don't think that we need given method, it should be possible to recursively call common awaitPrimaryReplica with timestamp = currentPrimaryReplicaWithOfflineHost.expirationTimestamp.tick().